packages/web/src/components/premium-content-purchase-modal/PremiumContentPurchaseModal.tsx

```diff
@@ -256,6 +266,13 @@ export const PremiumContentPurchaseModal = () => {
         ? PurchaseVendor.COINFLOW
         : PurchaseVendor.STRIPE
     })
+  const showGuestCheckout =
+    guestCheckoutEnabled &&
+    (!currentUser || (currentUser && !currentUser.handle)) &&
+    initialValues.guestEmail === ''
+  if (showGuestCheckout) {
+    dispatch(setPurchasePage({ page: PurchaseContentPage.GUEST_CHECKOUT }))
+  }
```

Comment:
This should be in a `useEffect`

Reply:
I also don't _love_ using a component lifecycle `dispatch` to reroute like this but reworking would probably be significant effort so this is fine

Reply:
yeaaa figured this would be the way that fits the current pattern